languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/DateCheckFilter.java

```diff
@@ -35,6 +36,11 @@ protected Calendar getCalendar() {
     return dateFilterHelper.getCalendar();
   }
 
+  @Override
+  protected String getErrorMessageWrongYear()  {
+    return "Dieses Datum stimmt nicht. Meinten Sie \"{currentYear}\"?";
```

Comment:
Perhaps: "Dieses Datum stimmt nicht mit dem Tag überein." (This date does not coincide with the day.)